File: solutions/kotlin/difference-of-squares/1/src/main/kotlin/DifferenceOfSquares.kt
class Squares(
    private val n: Int
) {
    fun sumOfSquares(): Int {
        return n * (n + 1) * (2 * n + 1) / 6
    }

    fun squareOfSum(): Int {
        return n * n * (n + 1) * (n + 1) / 4
    }

    fun difference(): Int {
        return squareOfSum() - sumOfSquares()
    }
}

File: solutions/typescript/difference-of-squares/1/difference-of-squares.ts
export class Squares {
  count: number
  
  constructor(count: number) {
    this.count = count
  }

  get sumOfSquares(): number {
    var res = 0
    for (let i = 1; i <= this.count; i++) {
      res += i ** 2
    }
    return res
  }

  get squareOfSum(): number {
    var res = 0
    for (let i = 1; i <= this.count; i++) {
      res += i
    }
    return res ** 2
  }

  get difference(): number {
    return this.squareOfSum - this.sumOfSquares
  }
}
